packages/codemods/fastify/4/url-params-optional/src/index.ts

```diff
@@ -0,0 +1,33 @@
+import type { API, FileInfo, Options } from "jscodeshift";
+
+export default function transform(file: FileInfo, api: API, options?: Options) {
+  const j = api.jscodeshift;
+  const root = j(file.source);
+  let dirtyFlag = false;
+
+  // Find all fastify.get calls
+  root
+    .find(j.CallExpression, {
+      callee: {
+        object: { name: "fastify" },
+        property: { name: "get" },
+      },
+    })
+    .forEach((path) => {
+      const firstArg = path.value.arguments[0];
+
+      // Ensure the first argument is a string literal
+      if (j.Literal.check(firstArg) && typeof firstArg.value === "string") {
+        const routePath = firstArg.value;
+
+        // Check if the route path matches the pattern '/posts/:id'
+        if (routePath === "/posts/:id") {
+          // Update the route path to '/posts/:id?'
+          firstArg.value = "/posts/:id?";
+          dirtyFlag = true;
+        }
+      }
```

Comment:
this will work only for this specific url, rest is not handled

Reply:
this is manual change and we might not need a codemod for this. this is not required.